lib/Sema/CSDiag.cpp

```diff
@@ -5372,6 +5382,137 @@ bool FailureDiagnosis::diagnoseCallContextualConversionErrors(
   return false;
 }
 
+// Offer a fix-it for using `subscript` instead of the operator
+// if the inputs are compatible.
+static bool diagnoseSubscriptMisuse(ApplyExpr *callExpr,
+                                    ConstraintSystem &CS,
+                                    FailureDiagnosis &FD) {
+  auto UDE = dyn_cast<UnresolvedDotExpr>(callExpr->getFn());
+  if (!UDE)
+    return false;
+
+  auto baseExpr = UDE->getBase();
+  if (baseExpr && UDE->getName().getBaseName() == "subscript") {
```

Comment:
This could be changed to `if (!baseExpr || UDE->getName().getBaseName() != "subscript") return false` to minimize indentation of the this function body.